android/src/main/java/com/reactlibrary/JudoReactNativeModule.kt

```diff
@@ -0,0 +1,669 @@
+                receipt?.let {
+                    transactionPromise?.resolve(receipt)
+                }
+            }
+            transactionPromise = null
+        }
+    }
+
+    // ------------------------------------------------------------
+    // MARK: Initializer
+    // ------------------------------------------------------------
+
+    init {
+        context.addActivityEventListener(listener)
+    }
+
+    // ------------------------------------------------------------
+    // MARK: SDK Methods
+    // ------------------------------------------------------------
+
+    @ReactMethod
+    fun invokeTransaction(options: ReadableMap, promise: Promise) {
+        getTransactionConfiguration(options)?.let {
+            startJudoActivity(it, promise)
+        }
+    }
+
+    @ReactMethod
+    fun invokeGooglePay(options: ReadableMap, promise: Promise) {
+        getGoogleTransactionConfiguration(options)?.let {
+            startJudoActivity(it, promise)
+        }
+    }
+
+    @ReactMethod
+    fun invokePaymentMethodScreen(options: ReadableMap, promise: Promise) {
+        getPaymentMethodsConfiguration(options)?.let {
+            startJudoActivity(it, promise)
+        }
+    }
+
+    private fun startJudoActivity(configuration: Judo, promise: Promise) {
+        transactionPromise = promise
+
+        val intent = Intent(currentActivity, JudoActivity::class.java)
+        intent.putExtra(JUDO_OPTIONS, configuration)
+
+        currentActivity?.startActivityForResult(intent, JUDO_PAYMENT_WIDGET_REQUEST_CODE)
+    }
+
+    // ------------------------------------------------------------
+    // MARK: Helper methods
+    // ------------------------------------------------------------
+
+    private fun getTransactionConfiguration(options: ReadableMap): Judo? {
+        val widgetType = getWidgetType(options)
+        return getJudoConfiguration(widgetType, options)
+    }
+
+    private fun getGoogleTransactionConfiguration(options: ReadableMap): Judo? {
+
+        val transactionMode = try {
+            options.getInt("transactionMode")
+        } catch (_: Exception) {
+            return null
+        }
+
+        val type = when (transactionMode) {
+            0 -> PaymentWidgetType.GOOGLE_PAY
+            else -> PaymentWidgetType.PRE_AUTH_GOOGLE_PAY
+        }
+
+        return getJudoConfiguration(type, options)
+    }
+
+    private fun getPaymentMethodsConfiguration(options: ReadableMap): Judo? {
+
+        val transactionMode = try {
+            options.getInt("transactionMode")
+        } catch (_: Exception) {
+            return null
+        }
+
+        val type = when (transactionMode) {
+            0 -> PaymentWidgetType.PAYMENT_METHODS
+            else -> PaymentWidgetType.PRE_AUTH_PAYMENT_METHODS
+        }
+
+        return getJudoConfiguration(type, options)
+    }
+
+    private fun getJudoConfiguration(type: PaymentWidgetType, options: ReadableMap): Judo? {
```

Comment:
can we avoid optionals here,
let's return a `Judo` object, and re-throw the exception to be handled by the caller:

```
    fun invokeGooglePay(options: ReadableMap, promise: Promise) {
        try {
        	val judo = getGoogleTransactionConfiguration(options)
			startJudoActivity(judo, promise)
        } catch (error: Error) {
        	promise.reject(error)
        }
    }

```

Reply:
Fixed